### PR TITLE
cluster: fix the bug that pd can not replicate region after deployed 

### LINKED ISSUE #3444: small cluster on single machine with low disk space doesn't schedule enough replicas
### What did you do?

I started a very simple, small cluster using tiup playground: `tiup playground nightly --kv=3 --tiflash=0`.

This cluster has `replication.max-replicas = 3`, and all other configuration is default except for these tikv configuration parameters:
```
[rocksdb]
max-open-files = 256

[raftdb]
max-open-files = 256

[storage]
reserve-space = 0
```

### What did you expect to see?

I expected that a table I created would have 3 replicas, one on each of the tikv nodes in the cluster. If that isn't possible, I'd expect the cluster to refuse to allow me to create tables or commit updates.

### What did you see instead?

The table has a single replica, scheduled to a single store. I think it would make more sense that if `max-replicas = 3`, and it is not possible to schedule enough replicas, it should not have been possible to create the table.

```
5.7.25-TiDB-v4.0.0-beta.2-2187-g1970a917c (root) [test]> select * from information_schema.tikv_store_status;
+----------+-----------------+-------------+------------------+-------+------------+----------+-----------+--------------+---------------+--------------+-------------+--------------+---------------+------------------+-------------+---------------------+---------------------+--------------+
| STORE_ID | ADDRESS         | STORE_STATE | STORE_STATE_NAME | LABEL | VERSION    | CAPACITY | AVAILABLE | LEADER_COUNT | LEADER_WEIGHT | LEADER_SCORE | LEADER_SIZE | REGION_COUNT | REGION_WEIGHT | REGION_SCORE     | REGION_SIZE | START_TS            | LAST_HEARTBEAT_TS   | UPTIME       |
+----------+-----------------+-------------+------------------+-------+------------+----------+-----------+--------------+---------------+--------------+-------------+--------------+---------------+------------------+-------------+---------------------+---------------------+--------------+
|        1 | 127.0.0.1:20160 |           0 | Up               | null  | 5.0.0-rc.x | 465.7GiB | 35.1GiB   |           25 |             1 |           25 |          25 |           25 |             1 | 72.8776791218132 |          25 | 2021-02-25 09:49:46 | 2021-02-25 09:53:56 | 4m10.232833s |
|        4 | 127.0.0.1:20162 |           0 | Up               | null  | 5.0.0-rc.x | 465.7GiB | 35.1GiB   |            0 |             1 |            0 |           0 |            0 |             1 |                0 |           0 | 2021-02-25 09:49:46 | 2021-02-25 09:53:56 | 4m10.232457s |
|        5 | 127.0.0.1:20161 |           0 | Up               | null  | 5.0.0-rc.x | 465.7GiB | 35.1GiB   |            0 |             1 |            0 |           0 |            0 |             1 |                0 |           0 | 2021-02-25 09:49:46 | 2021-02-25 09:53:56 | 4m10.232987s |
+----------+-----------------+-------------+------------------+-------+------------+----------+-----------+--------------+---------------+--------------+-------------+--------------+---------------+------------------+-------------+---------------------+---------------------+--------------+
3 rows in set (0.003 sec)

5.7.25-TiDB-v4.0.0-beta.2-2187-g1970a917c (root) [test]> select * from information_schema.tikv_region_status where table_name='t1';
+-----------+--------------------------------------+---------+----------+---------+------------+----------+----------+------------+----------------+---------------+---------------+------------+------------------+------------------+-------------------------+---------------------------+
| REGION_ID | START_KEY                            | END_KEY | TABLE_ID | DB_NAME | TABLE_NAME | IS_INDEX | INDEX_ID | INDEX_NAME | EPOCH_CONF_VER | EPOCH_VERSION | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE | APPROXIMATE_KEYS | REPLICATIONSTATUS_STATE | REPLICATIONSTATUS_STATEID |
+-----------+--------------------------------------+---------+----------+---------+------------+----------+----------+------------+----------------+---------------+---------------+------------+------------------+------------------+-------------------------+---------------------------+
|         2 | 7480000000000000FF3300000000000000F8 |         |       51 | test    | t1         |        0 |     NULL | NULL       |              1 |            25 |         13868 |       1643 |                1 |                0 | NULL                    |                      NULL |
+-----------+--------------------------------------+---------+----------+---------+------------+----------+----------+------------+----------------+---------------+---------------+------------+------------------+------------------+-------------------------+---------------------------+
1 row in set (0.005 sec)

5.7.25-TiDB-v4.0.0-beta.2-2187-g1970a917c (root) [test]> select * from information_schema.tikv_region_peers where region_id=2;
+-----------+---------+----------+------------+-----------+--------+--------------+
| REGION_ID | PEER_ID | STORE_ID | IS_LEARNER | IS_LEADER | STATUS | DOWN_SECONDS |
+-----------+---------+----------+------------+-----------+--------+--------------+
|         2 |       3 |        1 |          0 |         1 | NORMAL |         NULL |
+-----------+---------+----------+------------+-----------+--------+--------------+
1 row in set (0.003 sec)
```

### What version of PD are you using (`pd-server -V`)?
```
Release Version: v4.0.0-rc.2-459-g1b0dae46
Edition: Community
Git Commit Hash: 1b0dae4664105d63f292c583c59d2d07ab9bd45b
Git Branch: HEAD
UTC Build Time:  2021-02-24 08:07:12
```

### PR DESCRIPTION
**Pull request by bufferflies**
### What problem does this PR solve?
Fix #3444

### What is changed and how it works?
The bug was introduced by #3444
when TIDB deploy, the system will create 21 regions , pd can not replicate regions because of disk available radio . we hope the new system should work well even if the disk available radio do not satisfy the restriction。 so I modify the IsLowSpace's method, when the disk space exceed than 10G and the region count less than 30, pd should replicate

### Check List
Tests
- Manual test

Related changes
- Need to cherry-pick to the release branch

### Release note
- No release note

### CODE REVIEW

**Comment by nolouch** [comment]
cc @HunDunDM

File: server/core/store.go [hunk 1]
```diff
@@ -28,9 +28,11 @@ import (
...
 const (
 	// Interval to save store meta (including heartbeat ts) to etcd.
-	storePersistInterval = 5 * time.Minute
-	mb                   = 1 << 20 // megabyte
-	gb                   = 1 << 30
+	storePersistInterval   = 5 * time.Minute
+	mb                     = 1 << 20 // megabyte
+	gb                     = 1 << 30
+	initialMaxRegionCounts = 30          // exclude storage Threshold Filter when region less than 30
+	minimumSpace           = 10737418240 // 10GB
```
Comment (HunDunDM) [inline server/core/store.go:35]: The naming does not express the true meaning. It is best to add comments to these two parameters.
Use `1 << 30` to make it more readable.

File: server/core/store.go [hunk 2]
```diff
@@ -335,9 +337,16 @@ func (s *StoreInfo) AvailableRatio() float64 {
...
-// IsLowSpace checks if the store is lack of space.
+// IsLowSpace checks if the store is lack of space. not check if region count less
+// than initialMaxRegionCounts and available space more than minimumSpace
 func (s *StoreInfo) IsLowSpace(lowSpaceRatio float64) bool {
-	return s.GetStoreStats() != nil && s.AvailableRatio() < 1-lowSpaceRatio
+	if s.GetStoreStats() == nil {
+		return false
+	}
+	if s.regionCount < initialMaxRegionCounts && s.GetAvailable() > minimumSpace {
+		return false
```
Comment (HunDunDM) [inline server/core/store.go:347]: Add a link to the issue here for explanation.

**Comment by HunDunDM** [comment]
Please add a unit test to IsLowSpace in `store_test.go`.

**Review by HunDunDM — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**